src/nvim/option.c

```diff
@@ -5531,6 +5530,13 @@ int win_signcol_configured(win_T *wp, int *is_fixed)
   if (*scl == 'n'
       && (*(scl + 1) == 'o' || (*(scl + 1) == 'u'
                                 && (wp->w_p_nu || wp->w_p_rnu)))) {
+    if (!wp->w_buffer->b_signcols.valid) {
+      FOR_ALL_WINDOWS_IN_TAB(win, curtab) {
+        if (*win->w_p_stc != NUL) {
+          win->w_nrwidth_line_count = 0;
```

Comment:
Why is this done for all windows as opposed to just `wp`?

Reply:
Because the `buf_signcols()` call will validate `w_buffer->b_signcols.valid()`. Meaning that the next window handled in `update_screen()` showing the same buffer will not have it's statuscolumn invalidated.

Reply:
Perhaps moving this loop to the start of `buf_signcols()` itself would make that interaction more obvious.